feat: add C implementation for `stats/base/dists/uniform/stdev`

0PrashantYadav0 opened the pull request:
Resolves #3836

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for @stdlib/stats/base/dists/uniform/stdev along with relevant tests, benchmarks and examples

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3836

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: add C implementation for uniform standard deviation

stdlib-bot committed:
chore: update copyright years

0PrashantYadav0 commented:
/stdlib merge

0PrashantYadav0 committed:
Merge branch 'develop' into uniform/stdev

Signed-off-by: JoyBoy <144602492+0PrashantYadav0@users.noreply.github.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/stdev/include/stdlib/stats/base/dists/uniform/stdev.h:
```diff
@@ -0,0 +1,38 @@
+#endif
+
+/**
+* Returns the standard deviation of an uniform distribution.
```
```suggestion
* Returns the standard deviation of a uniform distribution.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/stdev/benchmark/benchmark.js:
```diff
@@ -31,6 +31,7 @@ var stdev = require( './../lib' );
 // MAIN //
 
 bench( pkg, function benchmark( b ) {
+	var len;
```
Already defined below:

```suggestion
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/stdev/README.md:
```diff
@@ -133,6 +133,97 @@ for ( i = 0; i < 10; i++ ) {
+
+#### stdlib_base_dists_uniform_stdev( a, b )
+
+Evaluates the [standard-deviation][standard-deviation] of a [uniform][uniform-distribution] distribution with `a` the minimum support and `b` the maximum support.
```
```suggestion
Returns the [standard deviation][standard-deviation] of a [uniform][uniform-distribution] distribution with parameters `a` (minimum support) and `b` (maximum support).
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/stdev/README.md:
```diff
@@ -133,6 +133,97 @@ for ( i = 0; i < 10; i++ ) {
+        a = random_uniform( 0.0, 10.0 );
+        b = random_uniform( a, a+10.0 );
+        v = stdlib_base_dists_uniform_stdev( a, b );
+        printf( "a: %lf, b: %lf, h(X;a,b): %lf\n", a, b, v );
```
```suggestion
        printf( "a: %lf, b: %lf, SD(X;a,b): %lf\n", a, b, v );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/stdev/package.json:
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "@stdlib/stats/base/dists/uniform/stdev",
+  "name": "@stdlib/stats/base/dists/uniform/kurtosis",
```
Some of the changes in this file need to be reverted.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/stdev/src/main.c:
```diff
@@ -0,0 +1,43 @@
+*
+* @param n   minimum support
+* @param p   maximum support
+* @return    stdev
```
```suggestion
* @return    standard deviation
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/stdev/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,70 @@
+	min = new Float64Array( len );
+	max = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		min[ i ] = ( randu()*10.0 );
```
This benchmark should be updated to use `uniform` instead of `randu`, similar to how `benchmark.js` looks now.

Planeshifter reviewed (changes requested):
Left a few comments that should be addressed; hopefully should be straightforward to resolve. Afterwards, this PR should be ready to land!

0PrashantYadav0 committed:
fix: changes according to given code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 committed:
docs: update README and comments for uniform distribution standard deviation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@Planeshifter Made all the changes you asked for please review and merge. Thank you.

Planeshifter reviewed (approved):
LGTM. Thanks!
